src/dawn_native/CommandBuffer.cpp

```diff
@@ -44,14 +44,23 @@ namespace dawn_native {
             if (uint64_t(location.x) + uint64_t(location.width) >
                     (static_cast<uint64_t>(texture->GetWidth()) >> level) ||
                 uint64_t(location.y) + uint64_t(location.height) >
-                    (static_cast<uint64_t>(texture->GetHeight()) >> level)) {
+                    (static_cast<uint64_t>(texture->GetHeight()) >> level) ||
+                uint64_t(location.z) + uint64_t(location.depth) >
```

Comment:
After [1] an additional check for array slice will be required too.

Reply:
New validation should also be covered by a [CopyCommandValidationTest](https://github.com/google/nxt-standalone/blob/master/src/tests/unittests/validation/CopyCommandsValidationTests.cpp).

Reply:
Done